Quickref.md

```diff
@@ -0,0 +1,168 @@
+[ ]   fmap ::   (a -> b) -> Behavior a -> Behavior b
+[ ]   ffor :: Behavior a ->   (a -> b) -> Behavior b
+[ ]   <*> :: Behavior (a -> b) -> Behavior a -> Behavior b
+[ ]   >>= :: Behavior a -> (a -> Behavior b) -> Behavior b
+[ ]   <> :: Monoid a => Behavior a -> Behavior a -> Behavior a
+-- ... plus many more due to typeclass membership
+
+-- Behavior to Behavior via monadic value
+[S]   sample :: Behavior a -> m a
+[ ]   pull   ::               m a -> Behavior a
+      -- Note supplied value is in [S] context
+```
+
+## Functions producing Dynamic
+
+```haskell
+-- Trivial Dynamic
+[ ]   constDyn ::                                  a            ->    Dynamic a
+
+-- Construct Dynamic from Event
+[H]   holdDyn       ::                             a -> Event a -> m (Dynamic a)
+[H]   foldDyn       :: (a -> b ->           b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynMaybe  :: (a -> b ->     Maybe b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynM      :: (a -> b -> m'        b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynMaybeM :: (a -> b -> m' (Maybe b)) -> b -> Event a -> m (Dynamic b)
+      -- Note m' supplies [H] context
+[H]   count :: Num b => Event a -> m (Dynamic b)
+[H]   toggle :: Bool -> Event a -> m (Dynamic Bool)
+
+-- Transform Dynamic to Dynamic using function
+[H]   forDyn  ::   Dynamic a ->  (a -> b) -> m (Dynamic b)
+[H]   mapDyn  :: (a ->    b) -> Dynamic a -> m (Dynamic b)
+[H]   mapDynM :: (a -> m' b) -> Dynamic a -> m (Dynamic b)
+      -- Note m' supplies [S] context
+[H]   splitDyn :: Dynamic (a, b) -> m (Dynamic a, Dynamic b)
+
+-- Combine multiple Dynamics
+[H]   mconcatDyn :: Monoid a => [Dynamic a] -> m (Dynamic a)
+[H]   distributeDMapOverDyn :: GCompare k => DMap (WrapArg Dynamic k) -> m (Dynamic (DMap k))
+[H]   combineDyn :: (a -> b -> c) -> Dynamic a -> Dynamic b -> m (Dynamic c)
+
+-- Efficient one-to-many fanout
+[ ]   demux      :: Ord k => Dynamic k -> Demux k
+[H]   getDemuxed ::  Eq k =>              Demux k -> k -> m (Dynamic Bool)
+
+-- Dynamic to Dynamic, removing updates w/o value change
+[ ]   nubDyn :: Eq a => Dynamic a -> Dynamic a
+
+-- Dynamic to identical Dynamic with debug trace.
+[ ]   traceDyn     :: Show a => String -> Dynamic a -> Dynamic a
+[ ]   traceDynWith ::    (a -> String) -> Dynamic a -> Dynamic a
+```
+
+## Flattening functions
+
+These functions flatten nested types such as Event-of-Event, Behavior-of-Event, Event-of-Behavior etc, removing the outer wrapper.
+
+For Events, the returned Event fires whenever the latest Event supplied by the wrapper fires.  There are differences in how the functions handle *coincidences* -- situations where the old or new Event fires at the same instant that we switch from old to new.  In some cases, the output Event tracks the old Event at the instant of switchover, while in other cases it tracks the new Event.
+
+```haskell
+-- Flatten Event-of-Event to monadic Event.  New Event is used immediately.
```

Comment:
I think we can drop the "to monadic Event" part - perhaps it's better to think about it as flattening starting at a particular point in time.